### ISSUE TITLE
Create stream listener for server

### ISSUE THREAD

**Issue by TheMysteriousVincent** [TheMysteriousVincent/armuntef#1]
We have to add some kind of server stream listener to clearly start and shutdown the ArmA 3 server.

**Comment by Ragebone** [comment]
Why and how exactly do you mean that?

Stream listener for retrieving errors, or knowing when to stop?

A while back #shutdown just stopped working for me, never figured out why.

Used autotest to start the tests, initiated by a build config in intelli j with arma as external tool and a java project that outputs the rpt into the console since windows  arma lacks the stream.

I'm up for discussion if you haven't noticed : )

**Comment by TheMysteriousVincent** [comment]
I guess starting an ArmA 3 server works fine at all. But when it comes to CI testing, we need to get the exact status of the unit testing results because we can not set an error status on the ArmA 3 server.

A CI needs an error status to whether know if the unit tests where successful or not (see TravisCI, for example).

The stream listener would then check the stream input and returns an exit status if the unit test fails and returns a normal `0` status if all unit tests where successful.

**Comment by Ragebone** [comment]
Yes, the RPT is basically send to stdOut on linux, but at least a year or so ago, the 3 Streams that windows applications provide did only return garbage or nothing directly usefull.
Except the normal Server Console, i thought about ex-filtrating such messages through ingame chat-messages that get printed to the console but i havn't pursued that approach.

Since it is a mod, is it? did i overlook that?
Extensions are probably the way to go : )

**Comment by TheMysteriousVincent** [comment]
It is just a simple mission framework, though. Not a mod. It is simple to deploy then.

**Comment by TheMysteriousVincent** [comment]
`Extensions are probably the way to go : )` Why should we make things more complicated than it could be. A simple listener is easy to write and needs definetely less testing than a shitty extension for ArmA 3.

**Comment by Ragebone** [comment]
sorry, my bad, didn't look at it long enough and thought it was supposed to be a Mod.
And a old habit of externalizing as much as possible.

Please define "Simple listener" a bit more in depth since the topic only sets the context of starting and shutting down the server.

I'd expect such a "listener" to be for extracting Error States somewhere from arma due to the platform differences, complications and maybe handling shutting down after the test, in case Arma fails at that.
And how do you plan on sending or getting events or messages to the Listener ? 

Though i think that for the only purpose of starting and shutting down,  the -autotest launch-param should work well ! ?
I don't remember if it also prints the test case mission results to the console, which would be great since that should be easily accessible on all platforms.

**Comment by TheMysteriousVincent** [comment]
Actually, you are right. I didn't read the parameter `-autoTest`. Then the listener would be irrelevant too. Thats good at all. I might create a docker image for testing purposes and then we could create fitting (e.g. TravisCI) files.
